x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -136,32 +149,127 @@ private void triggerTasks() {
                 return;
             }
             LOGGER.info("triggering scheduled [ML] maintenance tasks");
-            executeAsyncWithOrigin(client,
-                ML_ORIGIN,
-                DeleteExpiredDataAction.INSTANCE,
-                new DeleteExpiredDataAction.Request(deleteExpiredDataRequestsPerSecond, TimeValue.timeValueHours(8)),
-                ActionListener.wrap(
-                    response -> {
-                        if (response.isDeleted()) {
-                            LOGGER.info("Successfully completed [ML] maintenance tasks");
-                        } else {
-                            LOGGER.info("Halting [ML] maintenance tasks before completion as elapsed time is too great");
-                        }
-                    },
-                    e -> LOGGER.error("An error occurred during maintenance tasks execution", e)));
-            auditUnassignedMlTasks(clusterService.state());
+
+            // Step 3: Log whether the tasks have finished
+            ActionListener<AcknowledgedResponse> finalListener = ActionListener.wrap(
+                unused -> LOGGER.info("Completed [ML] maintenance tasks"),
```

Comment:
I wonder if it's best to remove this line and leave the success logging to the individual steps.  This will get logged when the last step succeeds, even if earlier steps failed.  As long as every other step logs successful completion this doesn't really add anything.